pages/price-feeds/use-real-time-data/ton.mdx

```diff
@@ -124,6 +124,112 @@ This code snippet does the following:
+  "P"-->>"M": op 5 parse_price_feed_updates (prices)
+  note over "M": Master accepts and processes the transaction
+```
+
+- Related code (GitHub):
+  - [master.fc: entry for Pyth message](https://github.com/evaafi/contracts/blob/d9138cb24f03b53522774351aceb38c51a047eee/contracts/master.fc#L121)
+  - [Pyth proxy: supply_withdraw (TON) in master.fc](https://github.com/evaafi/contracts/blob/d9138cb24f03b53522774351aceb38c51a047eee/contracts/master.fc#L192-L211)
+  - [Pyth proxy: liquidate (TON) in master.fc](https://github.com/evaafi/contracts/blob/d9138cb24f03b53522774351aceb38c51a047eee/contracts/master.fc#L171-L190)
+
+
+## Pyth proxy: Error handling
+In the Pyth proxy pattern, when an error occurs (i.e., Pyth cannot process the request and sends a `response_error` with op 0x10002), the error report is sent directly back to the user who initiated the transaction, not to a contract. This is different from the on-chain getter pattern, where the error is returned to the EVAA Master contract for further handling and potential refund logic. In the proxy case, the user receives the error response from the Pyth contract, including the error code and the original query ID, allowing the user to identify and handle the failure on their side.
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant "U" as "User"
+  participant "P" as "Pyth Contract"
+
+  "U"->>"P": request
+  "P"-->>"U": response_error (op 0x10002) with error_code and query_id
+```
+
+
+## Pyth onchain-getter: Success
+
+```mermaid
+sequenceDiagram
+  autonumber
+  participant "U" as "User"
+  participant "JW" as "Jetton Wallet"
+  participant "M" as "EVAA Master"
+  participant "P" as "Pyth Contract"
+
+  "U"->>"JW": transfer with forward_payload
+  note right of "U": op: liquidate_master or supply_withdraw_master_jetton
+  "JW"->>"M": transfer_notification
+  "M"->>"P": op 5 parse_price_feed_updates<br/>+ update_data + target_feeds<br/>+ op_payload(liquidate_master_jetton_process / supply_withdraw_master_jetton)
+  "P"-->>"M": op 5 parse_price_feed_updates<br/>(prices, pyth_sender = M)
+  note over "M": Master accepts and processes the transaction
+```
+
+- Related code (GitHub):
+  - [Entry: master.fc (pyth_parse_price_feed_updates)](https://github.com/evaafi/contracts/blob/d9138cb24f03b53522774351aceb38c51a047eee/contracts/master.fc#L121)
+  - [Onchain-getter branches: liquidate jetton and supply_withdraw jetton](https://github.com/evaafi/contracts/blob/d9138cb24f03b53522774351aceb38c51a047eee/contracts/master.fc#L131-L167)
+  - [Request to Pyth (liquidate jetton)](https://github.com/evaafi/contracts/blob/d9138cb24f03b53522774351aceb38c51a047eee/contracts/core/master-liquidate.fc#L728-L742)
+  - [Request to Pyth (supply_withdraw jetton)](https://github.com/evaafi/contracts/blob/d9138cb24f03b53522774351aceb38c51a047eee/contracts/core/master-supply-withdrawal.fc#L446-L461)
+
+## Pyth onchain-getter: Pyth error
```

Comment:
make it ### as all these are coming in the navigation bar.

Reply:
ok

Reply:
fixed